### DISCUSSION TITLE
Questions about running "./UFS_forcing_ensemble_generator.py"

### DISCUSSION THREAD

**Discussion by wzzheng90** [NCAR/ccpp-scm#382] [Cases, Initial Conditions, Forcing]
I am trying to run "./UFS_forcing_ensemble_generator.py" for your examples in [section 5.5 on the Tech Doc], and got some errors:

1) No the staged UWM RTs located at:   Hera      /scratch1/BMC/gmtb/CCPP-SCM/UFS_RTs
          Could you please put these examples to Hera  (/scratch1/BMC/gmtb/CCPP-SCM/UFS_RTs)?   Thanks.

2) Testing on Cheyenne, according to 5.5.4 Example 1: UFS-replay for single point
     ./UFS_forcing_ensemble_generator.py -d /glade/scratch/epicufsrt/GMTB/CCPP-SCM/UFS_RTs/control_c192/ -sc --C_RES 192 -dt 360 -n control_c192 -lons 300 -lats 

   The error messages show as follows:
            (a) ModuleNotFoundError: No module named 'ESMF'
            (b)  No such file or directory:     ~/ccpp-scm/scm/data/processed_case_input/control_c192_n000_SCM_driver.nc

      How to load 'ESMF' and obtain "control_c192_n000_SCM_driver.nc"?

Thank you very much!

Weizhong

**Comment by dustinswales** [comment]
@wzzheng90 Sorry to hear that you are running into issues.
Two quick comments. 1) In section 5.5.1 of the SCM users guide there is guidance on how to create the python environment needed to run the UFS_forcing_ensemble_generator.py script. There is a YAML file in the repository, environment-ufsreplay.yml, that lists the python dependencies. 2) I will upload the sample RT data to Hera (I uploaded it to Cheyenne, but forgot to do it on Hera)

**Reply by wzzheng90** [reply]
Thank you very much for your response.  Just one thing to add:  Yes, I created the python environment (environment-ufsreplay.yml) according to section 5.5.1 before I ran the UFS_forcing_ensemble_generator.py script.  I will try it on Hera again after you load the sample RT data.  
-- Weizhong

**Reply by dustinswales** [reply]
@wzzheng90 Sorry this is still not working. One more question. You need to create the environment **once**, but you also need to activate the environment at **each** new session

**Comment by wzzheng90** [comment]
Hi, Dustin,

Yes, I activate the environment at *each* new session on Cheyenne. I will
try it on Hera again after the data is available.
Thanks,

Weizhong

**Comment by wzzheng90** [comment]
@dustinswales 

Hi, Dustin,

Thank you very much for your loading the sample RT data on Hera.
On Hera, I cleaned the python environment (environment-ufsreplay.yml) on my home directory and created it again. Then, I run "./UFS_forcing_ensemble_generator.py" with your example, and it works now!

Just one question about this message after I run "./UFS_forcing_ensemble_generator.py" as follows:
          "Forcing should not be interpolated in time. Rather, forcing should held constant at their current values until the next forcing interval is reached."

I was supposed that the forcing should be changed in time.  Why it's constant?   Any options needed to set up?
Thanks again for your time and efforts on my questions.

Weizhong

**Reply by dustinswales** [reply]
@wzzheng90 Great to hear that it's working!

I probably should've removed that print statement, it's not needed. Here, when using the UFS standard output files to derive the forcings, the forcings are constant throughout the timestep. This is simple and could be extended to use something more robust (i.e Runge-Kutta)

**Reply by wzzheng90** [reply]
@dustinswales  I see.  Thanks again.
